File: patterns/c-patterns/pattern101.c
// Pattern 101: ...

//    *
//   **
//  * *
// *  *
//  * *
//   **
//    *

#include <stdio.h>

int main()
{
    int n = 7;
    int x = n / 2 + 1;

    int i, j;

    for (i = 1; i <= n; i++)
    {
        for (j = 1; j <= n / 2 + 1; j++)
        {
            if (j == x || j == n / 2 + 1)
                printf("*");
            else
                printf(" ");
        }
        if (i <= n / 2)
            x--;
        else
            x++;
        printf("\n");
    }
    return 0;
}

File: patterns/javascript-patterns/pattern101.js
// Pattern 101: ...

//    *
//   **
//  * *
// *  *
//  * *
//   **
//    *

let n = 7;
let x = Math.floor(n / 2) + 1;

for (let i = 1; i <= n; i++) {
    let row = '';

    for (let j = 1; j <= Math.floor(n / 2) + 1; j++) {
        if (j === x || j === Math.floor(n / 2) + 1) {
            row += '*';
        } else {
            row += ' ';
        }
    }

    if (i <= Math.floor(n / 2)) {
        x--;
    } else {
        x++;
    }

    console.log(row);
}
